formulaire/src/main/resources/public/ts/directives/question/question-type/question-type-singleanswer-radio/question-type-singleanswer-radio.directive.ts

```diff
@@ -0,0 +1,70 @@
+import {Directive, ng} from "entcore";
+import {FormElement, FormElements, Question} from "@common/models";
+import {I18nUtils} from "@common/utils";
+import {Direction} from "@common/core/enums";
+import {RootsConst} from "../../../../core/constants/roots.const";
+
+interface IQuestionTypeSingleanswerRadioProps {
+    question: Question;
+    hasFormResponses: boolean;
+    formElements: FormElements;
+    I18n: I18nUtils;
+    Direction: typeof Direction;
+}
+
+interface IViewModel extends ng.IController, IQuestionTypeSingleanswerRadioProps {
+    deleteChoice(index: number): Promise<void>;
+    isFormElementAfter(formElement: FormElement): boolean;
+}
+
+class Controller implements IViewModel {
+    question: Question;
+    hasFormResponses: boolean;
+    formElements: FormElements;
+    I18n: I18nUtils;
+    Direction: typeof Direction;
+
+    constructor(private $scope: IQuestionTypeSingleanswerRadioProps, private $sce: ng.ISCEService) {
```

Comment:
Du coup dans le constructor modifier le typage de $scope :
```suggestion
    constructor(private $scope: IQuestionTypeSingleanswerRadioScope, private $sce: ng.ISCEService) {
```